platform/SingletonPtr.h

```diff
@@ -118,15 +118,20 @@ struct SingletonPtr {
      * @returns
      *   A reference to the singleton
      */
-    T &operator*()
+    T &operator*() const
     {
         return *get();
     }
 
     // This is zero initialized when in global scope
-    T *_ptr;
-    // Force data to be 4 byte aligned
-    uint32_t _data[(sizeof(T) + sizeof(uint32_t) - 1) / sizeof(uint32_t)];
+    mutable T *_ptr;
+#if __cplusplus >= 201103L
+    // Align data appropriately
+    alignas(T) mutable char data[sizeof(T)];
```

Comment:
Wrong name, should be `_data`. 
Note that `std::aligned_storage<Size, Alignement>` may be used but it is actually more verbose and less readable than the current expression: `mutable typename std::aligned_storage<sizeof(T), alignof(T)>::type _data;` 🤦‍♂️

Reply:
Good spot. Also, I'm aware that this may break anyway when a C++11 switch gets thrown. It fails on ARMC5 which sets __cplusplus high enough, but does not implement `alignas`

I considered  `std::aligned_storage`, but afaict it may well not work as well - you're only passing it size, not the actual type, so it has presumably has to assume that any type >= 8 bytes might need 8-byte alignment. Whereas `alignas(type)` can check if there are any uint64_ts or whatever in there.

I do wonder if I'm missing something, because `std::aligned_storage` just seems pretty redundant when you have `alignas`. Maybe it came first.